components/form/CheckboxGroup.tsx

```diff
@@ -0,0 +1,164 @@
+import React from "react";
+import { useFormContext } from "react-hook-form";
+import { Label, HelperLabel } from "./Label";
+import { ErrorMessage } from "./ErrorMessage";
+import classNames from "classnames";
+import { FEEDBACK } from "@/utils/constant";
+
+type CheckboxProps = {
+  checked: boolean;
+  checkboxClassName: string;
+  containerClassName: string;
+  description: string;
+  disabled: boolean;
+  helperLabel: boolean;
+  helperLabelClassName: string;
+  label: string;
+  name: string;
+  size: string;
+  value: any;
+};
+
+export function Checkbox({
+  checked,
+  checkboxClassName,
+  containerClassName,
+  description,
+  disabled,
+  helperLabel,
+  helperLabelClassName,
+  label,
+  name,
+  size,
+  value,
+}: CheckboxProps) {
+  const iconName = {
+    checked: {
+      sm: "bg-checked-sm",
+      md: "bg-checked-md",
+    },
+    "disabled-checked": {
+      sm: "bg-checked-sm-disabled",
+      md: "bg-checked-md-disabled",
+    },
+  };
+
+  const { register } = useFormContext();
+
+  const iconClass =
+    (disabled && checked) || disabled ? "disabled-checked" : "checked";
+
+  return (
+    <>
+      <div className="flex relative">
+        <div
+          className={classNames(
+            "flex items-center",
+            { "pointer-events-none": checked },
+            containerClassName
+          )}
+        >
+          <input
+            checked={checked}
+            disabled={disabled}
+            id={`checkbox-${name}-${value.toString()}`}
+            className={`form-checkbox ${
```

Comment:
The logic here looks rough, we could simplify it by properly using the object notation logic for classNames.

Reply:
Noted